src/prefect/core/task.py

```diff
@@ -393,7 +393,7 @@ def __init__(
                         f"Invalid `retry_on` value {v!r}. Expected an exception subclass."
                     )
         else:
-            self.retry_on = None
+            self.retry_on = set([])
```

Comment:
Let's leave this as `None` please and make the type just include `Optional[...]`. 

For future reference this could just be `set()`.